Homework-Assignment-Collecties-Lussen

mjerta opened the pull request:
Beste lezer,

Ik heb alles goed doorgenomen. Zelf vond ik het wel aardig gaan. Ik heb bij de bonus opdracht nog wat extra validatie toegevoegd.
Ik heb wel een vraagje:

Hoe kan ik met het instantiëren van de HashSet ook de object in een interface type stoppen(ik weet niet of ik dat goed benoem btw) zoals hieronder.

Dus dat je een HashSet in een Set zet zoals ik ook met de Hasmap  heb gedaan.
1. Map<Integer, String> numericAlpha = new HashMap<>();

Ik wou dus een HashSet in een Set zetten. Ik kwam daarmee met een aantal foutmeldingen waarbij ik het wel kon oplossen. maar dan had ik op een gegeven moment: zoiets als
Set<Integer> secretnumber =   (Hashmap<Integer> )randomnumbergenerator();

Ik kwam er nog niet helemaal met de syntax.

Ik kijk uit naar de feedback.

mjerta committed:
in progress

mjerta committed:
fiFinshed the translator assignment

mjerta committed:
Added the first 2 methods

mjerta committed:
Added the bonus assignment

- Take note of the extra while loop.
- This will only break if the entered value is rightly formatted.
- I put my own explination of most  pieces of code in comments.

smallproject commented on src/Translator.java:
```diff
@@ -0,0 +1,24 @@
+    return numericAlpha;
+  }
+
+  public void setNumericAlpha(Map<Integer, String> numericAlpha) {
```
volgs mij is deze in niet gebruikt

mjerta replied:
Ik heb ze zojuist weggehaald.

smallproject commented on src/Main.java:
```diff
@@ -1,8 +1,37 @@
+                System.out.println("Type een cijfer in van 0 t/m 9");
+                int number = scanner.nextInt();
+                scanner.nextLine();
+                if(number < 10 ) {
```
ik ben benieuwd wat gaat hier gebeuren als je min numbers gebruikt, maar misschien is het niet nodig

mjerta replied:
Ik heb een extra conditie toegevoegd. Negatieve getallen worden nu als invalide beschouwd.

smallproject commented on src/Bonus.java:
```diff
@@ -1,66 +1,102 @@
+
+    Random rand = new Random();
+    HashSet<Integer> numbers = new HashSet<>();
+    while (numbers.size() < 4) {
```
dit is wel mooi trucje

mjerta replied:
Yup! een makkelijk manier om de loop door te lopen en deze te verhogen met door `.add`.

smallproject commented on src/Bonus.java:
```diff
@@ -1,66 +1,102 @@
-        String guess = scanner.nextLine();
-        if (Objects.equals(guess, stringnumber)) {
-            System.out.println("gefeliciteerd je hebt het goed");
+    // I added a while loop that will only break out if are right formatted value has been entered
```
heb je de lus helemaal veranderd?

mjerta replied:
Ik heb hier een while lus en wat extra if statements voor gebruikt ook om te checken of er een valide nummer wordt gebruikt en geen letters worden gebruikt.

mjerta committed:
Replaced type & removed get and set methods

- Replaced hashset into set type.
- Removed the unessecary get and set methods.

mjerta committed:
Add excondition to check negative value
